### PR TITLE
Update macOS scripts and support multiple types of GPA (new)

### LINKED ISSUE #71: Improved GPA calculator
* [x] Percent 
* [x] Unweighted (out of 4)
* [x] Weighted (Out of 5 depending on AP, HN, or CP)
* [x] Make grade calculations work with new types of GPA (so weighted and 4.0 GPA's update when you change assignments)
* [x] Year-to-date GPA (dependent on #73)

### LINKED ISSUE #83: Revise Mac install script
- Remove bit that clones code from github.com (Instead, assume they have forked and cloned it or downloaded a zip file and then are running the script from there)
- Then download dependencies
- Separate password
- Try to make script double-clickable
- Write better instructions for README

### PR DESCRIPTION
**Pull request by kdk1616**
This PR contains two features.
* Closes #83. Fix macOS install and start scripts, including removing Redis from the start script.
* Closes #71. Make function for toggling between weighted, unweighted, and % GPA.

### CODE REVIEW

**Comment by psvenk** [comment]
For some reason, the previous pull request with these changes was closed. For reference, below is the list of issues I found in that PR. I will check them off as they are fixed and tested.
- [x] GPA indicator should be merged with GPA type selector, not with term dropdown
- [x] A GPA of 4.0, 3.0, etc. should be displayed with at least one place after the decimal point
- [x] The non-percent GPA formats do not play well with "Calculated GPA"

File: public/js/buttonFunctions.js [hunk 1]
```diff
@@ -61,46 +61,50 @@ let resetTableData = function() {
...
       $("#" + currentTerm).css('padding', "5px 16px 5px 16px");
-      $("#" + currentTerm).html("Current Quarter GPA: "  + GPA + "<br>Calculated GPA: " + calcGPA);
-      document.getElementById('gpa_select').options[0].innerHTML = "Current Quarter GPA: "  + GPA + "<br>Calculated GPA: " + calcGPA;
-      document.getElementById('gpa_select').options[1].innerHTML = "Current Quarter GPA: "  + GPA + "<br>Calculated GPA: " + calcGPA;
+      $("#" + currentTerm).html("Current Quarter GPA: "  + GPA.percent + "<br>Calculated GPA: " + calcGPA.percent);
+      document.getElementById('gpa_select').options[0].innerHTML = "Current Quarter GPA: "  + GPA.percent + "<br>Calculated GPA: " + calcGPA.percent;
+      document.getElementById('gpa_select').options[1].innerHTML = "Current Quarter GPA: "  + GPA.percent + "<br>Calculated GPA: " + calcGPA.percent;
 
     } else {
+	    tableData.currentTermData.calcGPA = computeGPA(tableData.currentTermData.classes,termConverter.indexOf(currentTerm));
```
Comment (Ruborcalor) [inline public/js/buttonFunctions.js]: Shouldn't computeGPA take a single argument?

File: public/home.js [hunk 2]
```diff
@@ -674,34 +672,180 @@ let termsReset = {};
...
+                    $(`#q${i}, #q${i}_gpa`).html(
+                        "Q" + i + " GPA: " + tableData.terms["q" + i].GPA.percent
+                    );
+                }
+				selectedElem.html(
+                    quarterName + " GPA: " +  quarterData.GPA.percent
+                );
+			}
+        }
+      }
```
Comment (Ruborcalor) [inline public/home.js]: It would be nice to move these functions from home.js to calculate_grade.js

File: public/js/buttonFunctions.js (hunk 1 above)
Comment (Ruborcalor) [inline public/js/buttonFunctions.js]: Why does computeGPA take two arguments here? It's signature seems to account for a single argument.

line 177 calculate_grade.js
```
function computeGPA(classes) {
 let sum = 0; // Sum of classes' grades
 let count = 0; // Number of classes
 let fourSum = 0; // Sum of classes' grades on 4.0 scale
 let fiveSum = 0; // Sum of classes' grades on 5.0 scale
```

**Review by Ruborcalor — COMMENTED** [review]
Overall great work! Two minor comments.

File: public/js/buttonFunctions.js (hunk 1 above)
Comment (psvenk) [reply public/js/buttonFunctions.js]: I don't know; I just removed the second argument in 7aa8356.

**Comment by psvenk** [comment]
I have addressed every issue that I have identified, except for redesigning the cumulative GPA indicator so that it updates along with the quarters in showing percent, unweighted, or weighted instead of just showing all three always. It would also be nice to have a calculated cumulative GPA which uses calcGPA for each quarter (falling back to GPA if needed).

**Review by psvenk — APPROVED** [review]
Looks good to me

**Comment by kdk1616** [comment]
I approve!

**Pull request merged**